### PR TITLE
Into

### LINKED ISSUE #733: [CLEANUP] Some "Option<&str>" are remaining
They should be replaced by "Into<Option<&str>>" in order to uniformize our api.

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
Fixes #733.

cc @sdroege @EPashkin

### CODE REVIEW

File: src/dialog.rs [hunk 1]
```diff
@@ -15,9 +15,14 @@ use auto::DialogExt;
 use ResponseType;
 
 impl Dialog {
-    pub fn new_with_buttons<T: IsA<Window>>(title: Option<&str>, parent: Option<&T>,
-            flags: DialogFlags, buttons: &[(&str, ResponseType)]) -> Dialog {
+    pub fn new_with_buttons<'a, I: Into<Option<&'a str>>, T: IsA<Window>>(
+        title: I,
+        parent: Option<&T>,
```
Comment (EPashkin) [inline src/dialog.rs]: This don't changed on purpose?

**Comment by EPashkin** [comment]
Needed fix to check:
```diff
diff --git a/check_init_asserts b/check_init_asserts
index 09f3b5b1..eefece95 100755
--- a/check_init_asserts
+++ b/check_init_asserts
@@ -5,7 +5,7 @@ BEGIN { $e = 0 }
 while (/^(\N*)\V*fn\s+(\w+)\s*(<[^(]+>)?\s*(\([^{;]+)\{\N*\n^(\N*)$/gms) {
 	($modifiers, $name, $signature, $first_line) = ($1, $2, $4, $5);
 	unless ($modifiers =~ /^\s*(\/\/|(pub )?unsafe)/ ||
 	unless ($modifiers =~ /^\s*(\/\/|(pub )?unsafe)/ ||
-		$signature =~ /^\((&(\x27\w+ )?)?(mut )?self\b/ ||
+		$signature =~ /^\(\s*(&(\x27\w+ )?)?(mut )?self\b/ ||
 		$signature =~ /^\(\)/ ||
 		$first_line =~ /^\s*(
 			assert_initialized_main_thread |
```

File: src/dialog.rs (hunk 1 above)
Comment (GuillaumeGomez) [reply src/dialog.rs]: We didn't implement `Into` on `IsA` yet so not for now.

File: check_init_asserts [hunk 2]
```diff
@@ -5,7 +5,7 @@ BEGIN { $e = 0 }
 while (/^(\N*)\V*fn\s+(\w+)\s*(<[^(]+>)?\s*(\([^{;]+)\{\N*\n^(\N*)$/gms) {
 	($modifiers, $name, $signature, $first_line) = ($1, $2, $4, $5);
 	unless ($modifiers =~ /^\s*(\/\/|(pub )?unsafe)/ ||
-		$signature =~ /^\((&(\x27\w+ )?)?(mut )?self\b/ ||
+                $signature =~ /^\(\s*(&(\x27\w+ )?)?(mut )?self\b/ ||
```
Comment (EPashkin) [inline check_init_asserts]: Please, fix padding here.

**Comment by EPashkin** [comment]
@GuillaumeGomez Thanks and 👍 if CI passed.
IMHO better restart one of gtk3.22 to be sure.

**Comment by GuillaumeGomez** [comment]
I need to update our lgpl-docs repository first then I'll restart travis.

**Pull request merged**